ihm/multi_state_scheme.py

```diff
@@ -0,0 +1,225 @@
+        self.begin_state = begin_state
+        self.end_state = end_state
+        self.details = details
+        self.dataset_group = dataset_group
+        self.kinetic_rate = kinetic_rate
+        self.relaxation_time = relaxation_time
+        # The _assigned_to_scheme variable tracks whether the connectivity
+        # has been assigned to a scheme. This is to ensure that each
+        # connectivity is only assigned to a single scheme.
+        self._assigned_to_scheme = False
+
+    def set_assigned_to_scheme(self):
+        self._assigned_to_scheme = True
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+
+class KineticRate(object):
+    """A kinetic rate that can be assigned to a connectivity.
+    The kinetic rate could be either a transition_rate_constant or
+    an equilibrium_constant.
+    In case of an equilibrium_constant, the equilibrium_constant_unit and
+    equilibrium_constant_determination_method are required.
+
+    :param float transition_rate_constant: A transition rate constant
+      describing the exchange between two states. Unit: per second.
+    :param float equilibrium_constant: An equilibrium constant describing the
+      exchange between two states
+    :param str equilibrium_constant_determination_method:
+      The method how the equilibrium_constant was determined.
+      Options are: ['equilibrium constant is determined from population',
+      'equilibrium constant is determined from kinetic rates, kAB/kBA',
+      'equilibrium constant is determined from another method not listed']
+    :param str equilibrium_constant_unit: Unit of the equilibrium constant.
+      Depending on what the process described,a unit might be applicable or not
+    :param str details: Details on the kinetic rate.
+    :param dataset_group: The DatasetGroup used to determine the kinetic rate.
+    :type dataset_group: :class:`ihm.dataset.DatasetGroup`
+    :param external_file: External file containing measurement data
+      for the kinetic rate.
+
+    """
+
+    def __init__(self, transition_rate_constant=None,
+                 equilibrium_constant=None,
+                 equilibrium_constant_determination_method=None,
+                 equilibrium_constant_unit=None,
+                 details=None,
+                 dataset_group=None,
+                 external_file=None):
+        self.allowed_eq_const_det_methods = [
+            'equilibrium constant is determined from population',
+            'equilibrium constant is determined from kinetic rates, kAB/kBA',
+            'equilibrium constant is determined from another method not listed'
+        ]
+        self.transition_rate_constant = transition_rate_constant
+        self.equilibrium_constant = equilibrium_constant
+        self.equilibrium_constant_unit = equilibrium_constant_unit
+        # check whether the equilibrium_constant_determination_method is
+        # within the list of options if an equilibrium constant is given
```

Comment:
You can use the `_text_choice_property` utility function to handle this. Import it from `ihm.model`.